x/programs/rust/wasmlanche_sdk/src/host/mod.rs

```diff
@@ -83,7 +87,7 @@ pub fn host_program_invoke(call_ctx: &Context, method_name: &str, args: &[u8]) -
 /// Allocate memory into the module's linear memory
 /// and return the offset to the start of the block.
 #[no_mangle]
-pub fn alloc(len: usize) -> *mut u8 {
+pub extern "C" fn alloc(len: usize) -> *mut u8 {
```

Comment:
Rust does not have a stable ABI, and while it's unlikely to change for function signatures, there are no guarantees. What that means is external calls could break when calling the same "binary" (WASM in this case) compiled with different versions of Rust. `extern "C"` uses the C ABI, which is stable.